008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,1021 @@
+We implemented a function to return the valid range of possible food intakes given a list of active treaties and accepted request messages, and use this both to check if a newly proposed treaty is compatible with previously accepted treaties and to \textit{legally} take food. We do this as in Algorithm \ref{code:treatyRange}.
+% \Cref does not seem to work with algorithm components
+
+ \begin{algorithm}
+        \SetKwInOut{Input}{Input}
+        \SetKwInOut{Output}{Output}
+        \Input{List of active treaties \textbf{$T$}, RequestLeaveFood message \textbf{$M$}}
+        \Output{Maximum amount of food allowed: \textbf{max}}
+        \BlankLine
+        max $\gets$ food received \
+        \BlankLine
+        \ForEach{$t \in T$}
+        {
+        \uIf{\textup{agent state} $\in$ $t$ \textup{condition}}{
+            takeAmount $\gets$ convert $t$ request value
+            
+            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
+                max $\gets$ takeAmount
+            }
+
+        }
+        \BlankLine
+        \uIf{\textup{neighbour requested to leave food}}{
+            takeAmount $\gets$ convert $t$ request value
+            
+            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
+                max $\gets$ takeAmount
+            }
+        }            
+        }
+        \caption{Function to ensure that an agent never breaks any accepted treaties.}
+        \label{code:treatyRange}
+\end{algorithm}
+
+An agent then consumes the amount smaller or equal to the resulting upper-bound \textit{max} that maximizes his utility.
+
+
+\subsection{Proposing Treaties}\label{proposing_treaties}
+
+
+Treaty proposals are fundamentally linked to each agent's social motive, given that they work to the benefit of collective utility. The terms of each treaty must be adhered to by both entities that have agreed to it, resulting in a situation where both agents will benefit the same amount. Aligning this with agent strategies, a true narcissistic agent, which would go so far as to behave with the intention of causing detriment to others, would not make any decisions that might benefit its neighbouring agents. As a result of this, the narcissistic agent would not propose any treaties, as even though it might reap the rewards of it, it will never offer other agents any assistance. On the other end of the spectrum, an altruistic agent, that always makes decisions based on maximising other agents' utility, would not behave in a way that might benefit itself. This too has the effect of an altruist never proposing any treaties as although it might offer aid to other agents it goes against its self-sacrificing nature.
```

Comment:
```suggestion
Treaty proposals are fundamentally linked to each agent's social motive, given that they work to the benefit of collective utility. The terms of each treaty must be adhered to by both entities that have agreed to it, resulting in a situation where both agents will benefit the same amount. Aligning this with agent strategies, a true narcissistic agent, which would go so far as to behave with the intention of causing detriment to others, would not make any decisions that might benefit its neighbouring agents. As a result of this, the narcissistic agent would not propose any treaties, as even though it might reap the rewards of a treaty, it will never offer other agents any assistance. On the other end of the spectrum, an altruistic agent that always makes decisions based on maximising other agents' utility, would not behave in a way that might benefit itself. This too has the effect of an altruist never proposing any treaties, as although it might offer aid to other agents, this would go against its self-sacrificing nature.
```